pilot/pkg/model/service.go

```diff
@@ -1275,7 +1275,7 @@ func (s *Service) GetExtraAddressesForProxy(node *Proxy) []string {
 // GetAllAddressesForProxy returns a k8s service's extra addresses to the cluster where the node resides.
 // Especially for dual stack k8s service to get other IP family addresses.
 func (s *Service) GetAllAddressesForProxy(node *Proxy) []string {
-	if features.EnableDualStack && node.Metadata != nil && node.Metadata.ClusterID != "" {
+	if (features.EnableDualStack || features.EnableAmbient) && node.Metadata != nil && node.Metadata.ClusterID != "" {
```

Comment:
features.EnableAmbient  can violates EnableDualStack setting?

Reply:
I would rather say that dual-stack is enabled by default in ambient mode.

Reply:
It is a env, why cannot user turn it off

Reply:
I don't know, I didn't add that condition.